compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrDeclarationStorage.kt

```diff
@@ -662,8 +662,11 @@ class Fir2IrDeclarationStorage(
                     val setter = property.setter
                     // TODO: this checks are very preliminary, FIR resolve should determine backing field presence itself
                     if (property.isConst || (property.modality != Modality.ABSTRACT && (irParent !is IrClass || !irParent.isInterface))) {
-                        if (initializer != null || getter is FirDefaultPropertyGetter ||
-                            property.isVar && setter is FirDefaultPropertySetter
+                        if (initializer != null ||
+                            getter is FirDefaultPropertyGetter ||
+                            property.isVar && setter is FirDefaultPropertySetter ||
+                            property.isReferredViaField == true ||
+                            property.hasJvmFieldAnnotation
```

Comment:
Should there be some test that would be enabled with this additional condition `hasJvmFieldAnnotation`?

Reply:
It should, but as mentioned above in the TODO, it's rather better to extend these conditions/tests separately during resolution. I'd file an issue later, like "when/how to determine backing field presence during resolution". I'll remove this for now, since it's out of scope of PR anyway.

Reply:
Filed https://youtrack.jetbrains.com/issue/KT-41014